### PR TITLE
add header block pattern page

### PR DESCRIPTION
**Pull request by Fercas123**

### CODE REVIEW

File: site/docs/patterns/header-block.mdx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+title: Header block
+layout: DetailPattern
+tags:
+  - pattern
+aliases:
+  - /salt/patterns/header-block
+data:
+  resources:
+    [
+      { href: "https://storybook.saltdesignsystem.com/?path=/story/patterns-header-block--header-block", label: "Header block examples" },
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:11]: Broken link? I don't think we'll have specific examples either.

File: site/docs/patterns/header-block.mdx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+<Diagram
+  src="/img/patterns/header-block/hero-shot.png"
+  alt="Example header block. Displaying close action, status, preheader and description."
+  border
+/>
+
+## When to use
+
+Use this pattern to:
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:26]: This pattern is very similar to https://www.saltdesignsystem.com/salt/patterns/list-filtering, but seems documented slightly differently. Do we need to align?

File: site/docs/patterns/header-block.mdx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+<Diagram
+  src="/img/patterns/header-block/header-block-anatomy.png"
+  alt="A header block anatomy."
+  border
+/>
+
+### Layout
+
+Header block can be configured to be displayed using three different sizes, each differentiated by the size of the padding and the heading type. We recommend using an appropriately sized header block, relative to the container element it is embedded within.
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:54]: Header block is a pattern so can't be configured

File: site/docs/patterns/header-block.mdx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+As a rule of thumb, a large container in Salt uses padding size of `--salt-spacing-300`, a medium container uses `--salt-spacing-200`, and a small container uses `--salt-spacing-100`. For example, the Dialog component uses `--salt-spacing-300` padding between elements, therefore embedding a header block also using `--salt-spacing-300` maintains a consistent layout and ensures all elements align as expected.
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:56]: Is this documented anywhere? Foundations maybe?

File: site/docs/patterns/header-block.mdx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  border
+/>
+
+<ExampleContainer type="positive" customPillText="Best practices">
+Header block with `--salt-spacing-100` padding does not contain an accent bar. We recommend using this header block size within components where visual emphasis is not important. Consider the available space for an accent bar, relative to the size of the component.
+</ExampleContainer>
+
+## Layout variations
+
+A header block has a number of possible layout variations depending on the amount of information you want to communicate, the user workflow and the container it will sit in.
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:70]: This sentences seems like it needs adjusting to speak about header block as a pattern

File: site/docs/patterns/header-block.mdx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+### With accent bar and all text elements
+
+Display headings with an accent bar which draws attention to the heading, reinforces brand identity, and remains consistent with Salt Design System’s theming options.
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:74]: "Display headings", confused me. I was trying to decided whether this was display in terms of typography or a command.  I think the sentence needs updating

File: site/docs/patterns/header-block.mdx [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+<Diagram
+  src="/img/patterns/header-block/with-accent-bar-and-text.png"
+  alt="Dialog with a header block using padding --salt-spacing-300"
+  content="Dialog with a header block using padding --salt-spacing-300"
+  border
+/>
+
+### With accent bar, heading and pre-header only
+
+The dialog example demonstrates a step in a process as introduced by the pre-header and heading. A description is not necessary as the content of the dialog is sufficiently self-explanatory for the user to navigate and complete the form field instructions.
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:86]: Aren't they mostly dialog examples? Also it feels like we should focus on the actual layout variation in our intro

File: site/docs/patterns/header-block.mdx [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+<Diagram
+  src="/img/patterns/header-block/accent-bar-heading-preheader.png"
+  alt="Dialog with a header block using heading and pre header only."
+  content="Dialog with a header block using heading and pre header only."
+  border
+/>
+
+### With accent bar, heading and description only
+
+The drawer example uses a heading accompanied by a description that provides additional information on the status of the content. A contextual overview of the content is not required in the form of a pre-header, as the content itself indicates a summary of information.
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:97]: The spacing (whitespace between elements) in this file seems inconsistent after this point

File: site/docs/patterns/header-block.mdx [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+<ImageSwitcher
+  images={[
+    { src: "/img/patterns/header-block/close-button-only.png", alt: "" },
+    { src: "/img/patterns/header-block/close-button-only-spacing.png", alt: "" },
+  ]}
+  label="Show overlay content area"
+  caption="Overlay with a header block using padding --salt-spacing-100, and close button on display only"
+/>
+
+### With status
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:125]: I got to the end and realised this doesn't feel like it follows any of our previous patterns. This should be a lot more similar to contact details that it currently is.

File: site/docs/patterns/header-block.mdx [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+The drawer example uses a heading accompanied by a description that provides additional information on the status of the content. A contextual overview of the content is not required in the form of a pre-header, as the content itself indicates a summary of information.
+<Diagram
+  src="/img/patterns/header-block/header-block-in-drawer.png"
+  alt="Drawer with a header block using accent bar, heading and description."
+  border
+/>
+
+### Small container
+<ImageSwitcher
```
Comment (joshwooding) [inline site/docs/patterns/header-block.mdx:105]: No intro?

File: site/public/img/patterns/header-block/header-block-anatomy.png
Comment (joshwooding) [inline site/public/img/patterns/header-block/header-block-anatomy.png:1]: Is this anatomy using the correct annotation style?

File: site/docs/patterns/header-block.mdx (hunk 3 above)
Comment (Fercas123) [reply site/docs/patterns/header-block.mdx:50]: @bhoppers2008 "Header block can ~~be configured to~~ be displayed" ?

File: site/docs/patterns/header-block.mdx (hunk 4 above)
Comment (Fercas123) [reply site/docs/patterns/header-block.mdx:52]: @bhoppers2008 https://www.saltdesignsystem.com/salt/foundations/spacing#hierarchical-spacing?

Comment (Fercas123) [reply site/docs/patterns/header-block.mdx]: TODO: Foundations update required to talk about container spacing. will check with @amritadesmet

**Review by bhoppers2008 — DISMISSED** [review]

File: site/docs/patterns/header-block.mdx [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+This drawer example uses a heading accompanied by a description that provides additional information on the status of the content. A contextual overview of the content is not required in the form of a pre-header, as the content itself indicates a summary of information.
+<Diagram
+  src="/img/patterns/header-block/header-block-in-drawer.png"
+  alt="Dialog with header and description text elements."
+  content="Dialog with header and description text elements."
+  border
+/>
+
+## Small container
+This overlay example shows a header block using `--spacing-100` padding with a title and close button.
```
Comment (origami-z) [inline site/docs/patterns/header-block.mdx:102]: Maybe this is the only place not having `--salt` prefix for token

**Review by origami-z — DISMISSED** [review]

File: site/docs/patterns/header-block.mdx (hunk 11 above)
Comment (Fercas123) [inline site/docs/patterns/header-block.mdx:102]:
```suggestion
This overlay example shows a header block using `--salt-spacing-100` padding with a title and close button.
```

**Review by origami-z — APPROVED** [review]

**Review by amritadesmet — APPROVED** [review]

**Pull request merged**